src/ert/_c_wrappers/enkf/ecl_config.py

```diff
@@ -1,207 +1,91 @@
             )
-
-            c_ptr = self._alloc_full(
-                have_eclbase,
-                data_file,
-                grid,
-                refcase_default,
-                schedule_prediction_file,
+        if ConfigKeys.REFCASE in config_dict:
+            ecl_config_args["refcase_file"] = os.path.realpath(
+                config_dict[ConfigKeys.REFCASE]
             )
-            if grid is not None:
-                grid.convertToCReference(None)
-
-        if c_ptr:
-            super().__init__(c_ptr)
-        else:
-            raise RuntimeError("Internal error: Failed constructing EclConfig!")
-
-    def free(self):
-        self._free()
-
-    def getDataFile(self):
-        return self._get_data_file()
-
-    def setDataFile(self, datafile):
-        self._set_data_file(datafile)
-
-    def validateDataFile(self, datafile) -> UIReturn:
-        """@rtype: UIReturn"""
-        return self._validate_data_file(datafile)
-
-    # -----------------------------------------------------------------
-
-    def get_gridfile(self) -> str:
-        return self._get_gridfile()
-
-    def set_gridfile(self, gridfile):
-        self._set_gridfile(gridfile)
-
-    def validateGridFile(self, gridfile):
-        return self._validate_gridfile(gridfile)
-
-    def getGrid(self) -> EclGrid:
-        return self._get_grid()
-
-    def getRefcaseName(self):
-        return self._get_refcase_name()
-
-    def loadRefcase(self, refcase):
-        self._load_refcase(refcase)
-
-    def getRefcase(self) -> EclSum:
-        refcase = self._get_refcase()
-        if refcase is not None:
-            refcase.setParent(self)
-
-        return refcase
-
-    def validateRefcase(self, refcase):
-        return self._validate_refcase(refcase)
-
-    def hasRefcase(self):
-        """@rtype: bool"""
-        return self._has_refcase()
-
-    # -----------------------------------------------------------------
-
-    def getDepthUnit(self):
-        return self._get_depth_unit()
-
-    def getPressureUnit(self):
-        return self._get_pressure_unit()
-
-    # -----------------------------------------------------------------
-
-    def active(self):
-        """
-        Has ECLIPSE been configured?"
-        """
-        return self._active()
-
-    def getLastHistoryRestart(self):
-        return self._get_last_history_restart()
+        return cls(**ecl_config_args)
 
     @property
-    def num_cpu(self):
-        """
-        Returns numbers cpu to be used defined in a DATA file
-        """
-        return self._get_num_cpu()
+    def num_cpu(self) -> Optional[int]:
+        if not self._data_file:
+            return None
+        return get_num_cpu_from_data_file(self._data_file)
+
+    def _load_grid(self) -> Optional[EclGrid]:
+        ecl_grid_file_types = [
+            EclFileEnum.ECL_GRID_FILE,
+            EclFileEnum.ECL_EGRID_FILE,
+        ]
+        if not get_file_type(self._grid_file) in ecl_grid_file_types:
```

Comment:
:robot:

Reply:
not sure about this one - my hunch is, it should `X not in LIST` rather thatn `not X in LIST`?

Reply:
if that's it, then it's done

Reply:
that was it!